src/components/timeRangePicker/TimeRangePicker.tsx

```diff
@@ -0,0 +1,302 @@
+          severity="medium"
+          icon="chevronRight"
+        />
+        <DatepickerContainer ref={calendarWrapper}>
+          <IconButton
+            onClick={() => setIsCalendarActive((prev) => !prev)}
+            severity="medium"
+            icon="calendar"
+          />
+          <Datepicker>
+            {isCalendarActive && (
+              <Calendar
+                className="calendar-panel"
+                date={selectedDate}
+                weekStartsOn={1}
+                onChange={(date) => handleDateSelection(date)}
+              />
+            )}
+          </Datepicker>
+        </DatepickerContainer>
+      </Row>
+      <Row className="time-range-picker">
+        <Label name="Hours:" />
+        <UnifiedMultipleSelect size="small">
+          {DEFAULT_TIME_RANGE_OPTIONS.map((timeRangeOption) => (
+            <SelectButton
+              key={timeRangeOption.id}
+              name={timeRangeOption.id}
+              value={timeRangeOption.id}
+              labelText={`${timeRangeOption.start} - ${timeRangeOption.end}`}
+              onClick={() => setSelectedTimeRange(() => timeRangeOption)}
+              isActive={timeRangeOption.id === selectedTimeRange.id}
+            />
+          ))}
+          <SelectButton
+            key="custom"
+            name="custom"
+            value="custom"
+            labelText="custom"
+            onClick={() =>
+              setSelectedTimeRange((prev) => {
+                return {
+                  ...prev,
+                  id: "custom",
+                };
+              })
+            }
+            isActive={selectedTimeRange.id === "custom"}
+          />
+        </UnifiedMultipleSelect>
+        <CustomTimeRangeSelector isVisible={showCustomTimeRange}>
+          <div>From</div>
+          <input
+            type="time"
+            name="time-range-start"
+            value={selectedTimeRange.start}
+            onChange={(e) => {
+              e.persist();
+              setSelectedTimeRange((prev) => {
+                return {
+                  ...prev,
+                  start: e.target.value,
+                };
+              });
+            }}
+          />
+          <div>to</div>
+          <input
+            type="time"
+            name="time-range-end"
+            value={selectedTimeRange.end}
+            min={selectedTimeRange.start}
+            onChange={(e) => {
+              e.persist();
+              setSelectedTimeRange((prev) => {
+                return {
+                  ...prev,
+                  end: e.target.value,
+                };
+              });
+            }}
+          />
+          <Button severity="high" onClick={submit} size="sm">
+            Apply
+          </Button>
+        </CustomTimeRangeSelector>
+      </Row>
+    </>
+  );
+}
+
+const getMonday = (date: Date): Date => {
+  const result = new Date(date);
+
+  if (isMonday(result)) return result;
+  else return previousMonday(result);
```

Comment:
We could use ternary here

```suggestion
  return isMonday(result) ? result : previousMonday(result);
```